.buildkite/scripts/steps/artifacts/cloud.sh

```diff
@@ -50,7 +50,17 @@ jq '
   .resources.integrations_server[0].plan.integrations_server.version = "'$FULL_VERSION'"
   ' .buildkite/scripts/steps/cloud/deploy.json > "$DEPLOYMENT_SPEC"
 
-ecctl deployment create --track --output json --file "$DEPLOYMENT_SPEC" &> "$LOGS"
+function shutdown {
+  echo "--- Shutdown deployment"
+  # Re-fetch the deployment ID - if there's an error during creation the ID may not be set
+  CLOUD_DEPLOYMENT_ID=$(ecctl deployment list --output json | jq -r '.deployments[] | select(.name == "'$CLOUD_DEPLOYMENT_NAME'") | .id')
+  if [ -z "${CLOUD_DEPLOYMENT_ID}" ]; then
+    ecctl deployment shutdown "$CLOUD_DEPLOYMENT_ID" --force --track --output json > "$LOGS"
+  fi
+}
+trap "shutdown" EXIT
+
+ecctl deployment create --track --output json --file "$DEPLOYMENT_SPEC" > "$LOGS"
```

Comment:
This was the source of the error - we're registering the hook before now.  The build logs cover the failing case.